### PR TITLE
Improve Usability and Reliability of AI vs AI

### LINKED ISSUE #1767: Make AI vs AI Gameplay Work and Document Procedures
### Description of the task
We should in theory be able to run AI vs AI by running an ssl-gamecontroller, 2 full systems, and a standalone simulator. However we run into the following problems:
* blue side robots are stuck (edit: seems to be channel issue)
* halt causes robots to bounce around erratically
* not sure if gamecontroller correct tells the full system which side each team is defending

### Acceptance criteria

- [ ] Make AI vs AI work and record a nice gif 
- [ ] Document procedures for running it
- [ ] Test behaviour introduced by #1768

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* fix default values to dynamic params and fullsystemgui comboboxes
* remove "Use gamecontroller" as an option for team colour
* make sensorfusion detect vision reset

### Testing Done
added unit test

### Resolved Issues
resolves #1767

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/run_simulated_ai_vs_ai.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+THIS_SCRIPT_FILENAME=$(basename "$0")
+
+# The number of arguments passed to this script
+NUM_ARGS=$#
+
+if [ $NUM_ARGS -ne 1 ]; then
+  echo "Error: Incorrect number of arguments to script"
+  echo "Usage: $THIS_SCRIPT_FILENAME [interface_name]"
+  echo "Choose from the following interfaces:"
+  echo "`ifconfig`"
```
Comment (akhilveeraghanta) [inline src/software/run_simulated_ai_vs_ai.sh]: you can run `ls /sys/class/net` net here for a more pruned output

File: src/software/run_simulated_ai_vs_ai.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  exit 0
+fi
+
+bazel run //software:full_system -- --backend=WifiBackend --interface=$1 &
+bazel run //software:full_system -- --backend=WifiBackend --interface=$1 &
+bazel run //software/simulation:standalone_simulator_main -- --interface=$1 &
+xdg-open http://localhost:8081 &
```
Comment (akhilveeraghanta) [inline src/software/run_simulated_ai_vs_ai.sh]: what about launching the gamecontroller? can we do a wget here into the `/opt` folder and run it?

**Review by akhilveeraghanta — COMMENTED** [review]
lgtm, just two minor comments

File: src/software/run_simulated_ai_vs_ai.sh (hunk 2 above)
Comment (jonathanlew) [reply src/software/run_simulated_ai_vs_ai.sh]: I put it in tmp so I don't need `sudo` privileges

File: src/software/run_simulated_ai_vs_ai.sh (hunk 1 above)
Comment (jonathanlew) [reply src/software/run_simulated_ai_vs_ai.sh]: fixed

File: src/software/run_simulated_ai_vs_ai.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  echo "Runs AI vs AI with the standalone simulator"
+  echo "Usage: $THIS_SCRIPT_FILENAME [interface_name]"
+  echo "Choose from the following interfaces:"
+  echo "`ls /sys/class/net`"
+  exit 0
+fi
+
+bazel run //software:full_system -- --backend=WifiBackend --interface=$1 &
+bazel run //software:full_system -- --backend=WifiBackend --interface=$1 &
+bazel run //software/simulation:standalone_simulator_main -- --interface=$1 &
```
Comment (akhilveeraghanta) [inline src/software/run_simulated_ai_vs_ai.sh]: Here is the promised tmux method, you should be able to click on the windows

~~~bash
tmux new-session \; \
  set -g mouse on \; \
  send-keys "bazel run //software:full_system -- --backend=WifiBackend --interface=$1" C-m \; \
  split-window -h \; \
  send-keys "bazel run //software:full_system -- --backend=WifiBackend --interface=$1" C-m \; \
  split-window -v \; \
  send-keys "/tmp/ssl-game-controller_v2.4.0_linux_amd64" C-m \; \
  select-pane -t 0 \; \
  split-window -v \; \
  send-keys "bazel run //software/simulation:standalone_simulator_main -- --interface=$1" C-m \;

~~~

**Review by akhilveeraghanta — APPROVED** [review]

File: docs/getting-started.md [hunk 4]
```diff
@@ -171,14 +184,17 @@ Now that you're setup, if you can run it on the command line, you can run it in
...
 
 ## Debugging
 Debugging from the command line is certainly possible, but debugging in a full IDE is *really* nice (plz trust us). 
 
-### With CLion
+### Debugging with CLion
 Debugging in CLion is as simple as running the above instructions for building CLion, but clicking the little green bug in the top right corner instead of the little green arrow!
 
-### From The Command line
+### Debugging from the Command line
```
Comment (kelvinkoon) [inline docs/getting-started.md:197]: Small nit, we refer to it as `command-line` earlier. Non-blocking but wanted to point it out.

Comment (jonathanlew) [reply docs/getting-started.md:197]: Made https://github.com/UBC-Thunderbots/Software/issues/1854 to track it.

**Review by kelvinkoon — APPROVED** [review]
Nice work!

**Pull request merged**